#!/usr/bin/env python
PACKAGE = "rosaria"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Default values set to 0 will be set from robot defaults

gen.add("trans_accel", double_t, 0, "Translational acceleration (m/s^2)", 0, 0 ) #max from robot config
gen.add("trans_decel", double_t, 0, "Translational deceleration (m/s^2)", 0, 0 ) #max from robot config
gen.add("lat_accel"  , double_t, 0, "Lateral acceleration (m/s^2)"      , 0, 0 ) #max from robot config
gen.add("lat_decel"  , double_t, 0, "Lateral deceleration (m/s^2)"      , 0, 0 ) #max from robot config
gen.add("rot_accel"  , double_t, 0, "Rotational acceleration (rad/s^2)" , 0, 0 ) #max from robot config
gen.add("rot_decel"  , double_t, 0, "Rotational deceleration (rad/s^2)" , 0, 0 ) #max from robot config

#bounds are relitivly arbitrarily, feel free to increase them.
gen.add("TicksMM"    , int_t, 0, "Encoder ticks/mm"                            , 0, 0, 256)
gen.add("DriftFactor", int_t, 0, "Value in 1/8192 increments to be added or " + 
                                 "subtracted from the left encoder ticks in " +
                                 "order to compensate for tire differences."   , 0, -200, 200)
gen.add("RevCount"   , int_t, 0, "The number of differential encoder ticks " + 
                                 "for a 180-degree revolution of the robot."   , 0, 0, 32760)

exit(gen.generate(PACKAGE, "RosAria", "RosAria"))
